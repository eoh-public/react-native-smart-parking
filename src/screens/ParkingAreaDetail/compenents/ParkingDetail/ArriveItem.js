import React, { memo, useCallback } from 'react';
import { StyleSheet, TouchableOpacity } from 'react-native';

import { Colors } from '../../../../configs';
import Text from '../../../../commons/Text';
import { calcTime } from '../../../../utils/Converter/time';
import { formatMoney } from '../../../../utils/Utils';
import { TESTID } from '../../../../configs/Constants';

const ArriveItem = memo(({ time, price, onPress, id, selected, index }) => {
  const onPressItem = useCallback(() => {
    onPress && onPress({ id, time, price }, index);
  }, [id, index, onPress, price, time]);
  const timeWithFormat = calcTime(time, 'H', 'LT');

  return (
    <TouchableOpacity
      testID={TESTID.ARRIVE_ITEM}
      style={[
        styles.container,
        {
          borderColor: selected ? Colors.Primary : Colors.Gray4,
        },
      ]}
      activeOpacity={0.4}
      onPress={onPressItem}
    >
      <Text type="Body" style={styles.time} color={Colors.Gray9}>
        {timeWithFormat}
      </Text>
      <Text type="Label" color={Colors.Gray7}>
        {formatMoney(price)}
      </Text>
    </TouchableOpacity>
  );
});

export default ArriveItem;

const styles = StyleSheet.create({
  container: {
    width: 80,
    borderWidth: 1,
    borderColor: Colors.Gray4,
    paddingVertical: 8,
    alignItems: 'center',
    marginRight: 16,
  },
  time: {
    marginBottom: 4,
  },
});
