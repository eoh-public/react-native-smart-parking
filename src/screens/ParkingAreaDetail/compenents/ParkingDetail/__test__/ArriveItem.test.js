import ArriveItem from '../ArriveItem';
import React from 'react';
import renderer, { act } from 'react-test-renderer';
import moment from 'moment';

describe('Test ArriveItem', () => {
  const time = moment('2021-01-23T04:34:32.204043Z').utcOffset(0);
  const price = 20000;
  const obj = { time: time, price: price };
  let tree;

  test('render ArriveItem', () => {
    act(() => {
      tree = renderer.create(<ArriveItem {...obj} />);
    });
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
