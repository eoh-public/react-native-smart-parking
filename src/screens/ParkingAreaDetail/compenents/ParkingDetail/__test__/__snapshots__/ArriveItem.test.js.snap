// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ArriveItem render ArriveItem 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#E8E8E8",
      "borderWidth": 1,
      "marginRight": 16,
      "opacity": 1,
      "paddingVertical": 8,
      "width": 80,
    }
  }
  testID="ARRIVE_ITEM"
>
  <Text
    style={
      Array [
        Object {
          "color": "#262626",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 14,
          "lineHeight": 22,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "marginBottom": 4,
        },
      ]
    }
  >
    4:34 AM
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#8C8C8C",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 12,
          "lineHeight": 20,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {},
      ]
    }
  >
    20.000 Ä‘
  </Text>
</View>
`;
